File: src/main/java/za/ac/sun/grapl/domain/stack/operand/ConstantItem.java
package za.ac.sun.grapl.domain.stack.operand;

import za.ac.sun.grapl.domain.stack.OperandItem;

public class ConstantItem extends OperandItem {

    public ConstantItem(String id, String type) {
        super(id, type);
    }

    @Override
    public String toString() {
        return "CONST {value: " + id + ", type: " + type + "}";
    }

}

File: src/main/java/za/ac/sun/grapl/domain/stack/operand/ConstantItem.kt
package za.ac.sun.grapl.domain.stack.operand

import za.ac.sun.grapl.domain.stack.OperandItem

class ConstantItem(id: String?, type: String?) : OperandItem(id!!, type!!) {
    override fun toString(): String {
        return "CONST {value: $id, type: $type}"
    }
}